front-end/src/app/app.component.ts

```diff
@@ -86,57 +85,61 @@ export class AppComponent {
   doneArray: Task[] = [];
   isDoneArrayEmpty = true;
   shouldShowTaskArray = false;
-  id = -1;
 
   applyForm = new FormGroup({
-    title: new FormControl(''),
-    comment: new FormControl(''),
+    title: new FormControl('', { nonNullable: true }),
+    comment: new FormControl('', { nonNullable: true }),
   });
 
   constructor(
     private readonly taskService: TaskService,
     private snackBar: MatSnackBar
   ) {
-    this.taskService.getTasks().then((taskArrayList: Task[]) => {
-      this.taskArray = taskArrayList;
+    this.taskService.getTasks().subscribe((taskArrayList: Task[]) => {
+      taskArrayList.forEach((task) => {
+        if (task.isDone) {
+          this.doneArray.push(task);
+          this.isDoneArrayEmpty = false;
+        } else this.taskArray.push(task);
+      });
       this.shouldShowTaskArray = true;
     });
   }
 
   submitTask() {
-    if (this.applyForm.value.title != '') {
-      let newTask = new Task(
-        this.id++,
-        this.applyForm.value.title ?? '',
-        this.applyForm.value.comment ?? ''
-      );
-      this.taskService.submitTask(newTask);
-      this.applyForm.reset();
-      this.applyForm.value.title = '';
-      this.applyForm.value.comment = '';
+    if (this.applyForm.value.title) {
+      let newTask: Task = {
+        title: this.applyForm.value.title,
+        comment: this.applyForm.value.comment,
+        isDone: false,
+      };
+
+      this.taskService.submitTask(newTask).subscribe((task) => {
+        this.taskArray.push(task);
+        this.applyForm.reset();
+      });
     } else {
       this.snackBar.open("Task title can't be empty!", '', {
         duration: 2500,
       });
     }
   }
 
-  doneTask(id: number) {
+  doneTask(id: string) {
     this.isDoneArrayEmpty = false;
-
-    this.taskArray.forEach((e) => {
-      if (e.id === id) {
-        this.doneArray.push(e);
-        this.taskArray.splice(this.taskArray.indexOf(e), 1);
+    this.taskArray.forEach((task) => {
+      if (task._id === id) {
+        task.isDone = true;
+        this.taskService.doneTask(task._id).subscribe();
```

Comment:
just like the delete function, here you should also wait for the successful api call before executing the frontend logic